File: Java/EstruturaDeRepeticao/Exercicio42.java
// Faça um programa que leia uma quantidade indeterminada de números positivos e conte quantos deles estão nos seguintes intervalos: [0-25], [26-50], [51-75] e [76-100].
// A entrada de dados deverá terminar quando for lido um número negativo.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio42 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int intervalo1 = 0, intervalo2 = 0, intervalo3 = 0, intervalo4 = 0;

        while (true) {
            System.out.print("Digite um número (negativo para encerrar): ");
            int numero = scanner.nextInt();

            if (numero < 0) {
                break;
            }

            if (numero >= 0 && numero <= 25) {
                intervalo1++;
            } else if (numero >= 26 && numero <= 50) {
                intervalo2++;
            } else if (numero >= 51 && numero <= 75) {
                intervalo3++;
            } else if (numero >= 76 && numero <= 100) {
                intervalo4++;
            }
        }

        System.out.println("Quantidade de números no intervalo [0-25]: " + intervalo1);
        System.out.println("Quantidade de números no intervalo [26-50]: " + intervalo2);
        System.out.println("Quantidade de números no intervalo [51-75]: " + intervalo3);
        System.out.println("Quantidade de números no intervalo [76-100]: " + intervalo4);

        scanner.close();
    }
}

File: Python/EstruturaDeRepeticao/exercicio_42.py
"""
Faça um programa que leia uma quantidade indeterminada de números positivos e conte quantos deles estão nos seguintes intervalos: [0-25], [26-50], [51-75] e [76-100].
A entrada de dados deverá terminar quando for lido um número negativo.
"""

intervalos = {
    "[0-25]": 0,
    "[26-50]": 0,
    "[51-75]": 0,
    "[76-100]": 0
}

while True:
    numero = float(input("Digite um número positivo (ou um número negativo para encerrar): "))
    if numero < 0:
       break

    if 0 <= numero <= 25:
        intervalos["[0-25]"] += 1
    elif 26 <= numero <= 50:
        intervalos["[26-50]"] += 1
    elif 51 <= numero <= 75:
        intervalos["[51-75]"] += 1
    elif 76 <= numero <= 100:
        intervalos["[76-100]"] += 1

print("Quantidade de números nos intervalos:")
for intervalo, quantidade in intervalos.items():
    print(f"{intervalo}: {quantidade}")

